packages/core/contracts/financial-templates/common/FeePayer.sol

```diff
@@ -102,6 +102,9 @@ abstract contract FeePayer is AdministrateeInterface, Testable, Lockable {
 
         // Exit early if there is no collateral from which to pay fees.
         if (collateralPool.isEqual(0)) {
+            // Note: set the lastPaymentTime in this case so the contract is credited for paying during periods when it
+            // has no locked collateral.
+            lastPaymentTime = time;
```

Comment:
Note: this was a minor bug that existed in our existing implementation. Fees are not "credited" while the pfc == 0. I think this timestamp should be updated even if we return early cc @chrismaree @nicholaspai.

Reply:
Ah, so IIUC without this change, once its PfC > 0, the contract would be charged retrospectively for all of the time it spent at PfC=0.

Reply:
Yep, exactly.

Reply:
good catch!